controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -125,6 +127,27 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, errors.Wrap(err, "could not load host data")
 	}
+	// Fetch the HostData
+	hostData := &metal3v1alpha1.HostData{}
+	hostDataKey := client.ObjectKey{
+		Name:      host.Name,
+		Namespace: host.Namespace,
+	}
+	err = r.Get(ctx, hostDataKey, hostData)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		fmt.Println("Handle it")
+	}
+
+	// Host is being deleted, so we delete the finalizer from the hostData to allow its deletion.
+	if !host.DeletionTimestamp.IsZero() {
+		if controllerutil.ContainsFinalizer(hostData, hostDataFinalizer) {
```

Comment:
Is there any use case where Finalizer would not exist?

Reply:
What would or expected to happen when finalizer is manually deleted?

Reply:
it allows the hostData to be deleted even if BMH doesn't have deletion timestamp

Reply:
My concern was, if user deletes finalizer manually before host deletion, and when actual deletion happens it won't be able to remove manually deleted finalizer and next try to [update](https://github.com/metal3-io/baremetal-operator/pull/1099/files#diff-a69d9029388ab766ed36b32180145f52785a9d4a153775510dbddfa928a72e1cR146-R148) hostData, which should throw an error I believe. I think we need to do the update inside the finalizer contains condition

Reply:
It won't throw an error during update, only if the client fails to update the resource on the API. But, it makes sense to add the update to within that if, because, there is no point of updating the resource if no change was made.

Reply:
updated